sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/TransactionalBulkExecutor.java

```diff
@@ -0,0 +1,472 @@
+                            signal,
+                            itemsLeftSnapshot,
+                            this.operationContextText,
+                            getThreadInfo());
+                    }
+                }
+
+                this.dispose();
+            });
+    }
+
+    private Flux<CosmosBulkOperationResponse<TContext>> executeCore() {
+
+        // For transactional batches, batches are pre-constructed by the writer with proper partition key grouping.
+        // We just need to execute each pre-built batch.
+        Integer nullableMaxConcurrentCosmosPartitions = cosmosBulkExecutionOptions.getMaxConcurrentCosmosPartitions();
+        int maxConcurrentBatches = nullableMaxConcurrentCosmosPartitions != null ?
+            Math.max(256, nullableMaxConcurrentCosmosPartitions) : 256;
+
+        logDebugOrWarning("TransactionalBulkExecutor.executeCore with MaxConcurrentBatches: {}, Context: {}",
+            maxConcurrentBatches,
+            this.operationContextText);
+
+        return this.inputBatches
+            .publishOn(this.executionScheduler)
+            .onErrorMap(throwable -> {
+                logger.warn("{}: Error observed when processing input batches. Cause: {}, Context: {}",
+                    getThreadInfo(),
+                    throwable.getMessage(),
+                    this.operationContextText,
+                    throwable);
+                return throwable;
+            })
+            .flatMap(
+                cosmosBatch -> executeTransactionalBatch(cosmosBatch),
+                maxConcurrentBatches)
+            .subscribeOn(this.executionScheduler);
+    }
+
+    private Flux<CosmosBulkOperationResponse<TContext>> executeTransactionalBatch(CosmosBatch cosmosBatch) {
+        // Extract operations from the pre-built batch
+        List<CosmosItemOperation> operations = cosmosBatch.getOperations();
+        
+        if (operations == null || operations.isEmpty()) {
+            logger.trace("Empty batch received, Context: {}", this.operationContextText);
+            return Flux.empty();
+        }
+
+        String batchTrackingId = UUIDs.nonBlockingRandomUUID().toString();
+        PartitionKey partitionKey = cosmosBatch.getPartitionKeyValue();
+        
+        logDebugOrWarning(
+            "Executing transactional batch - {} operations, PK: {}, TrackingId: {}, Context: {}",
+            operations.size(),
+            partitionKey,
+            batchTrackingId,
+            this.operationContextText);
+
+        // Create SinglePartitionKeyServerBatchRequest with atomic batch semantics
+        final SinglePartitionKeyServerBatchRequest request = SinglePartitionKeyServerBatchRequest.createBatchRequest(
+            partitionKey,
+            operations,
+            this.effectiveItemSerializer);
+        request.setAtomicBatch(true);
+        request.setShouldContinueOnError(false);
```

Comment:
Worth checking with backedn folks what this means when setAtomicBatch == true - maybe this is somethign we can optimize to implement retry polciies for soem of the write startegies like Delete where 404 Not Found should be silently ignored.